stl/inc/algorithm

```diff
@@ -2797,6 +2797,253 @@ namespace ranges {
     };
 
     _EXPORT_STD inline constexpr _Fold_right_last_fn fold_right_last{_Not_quite_object::_Construct_tag{}};
+
+    class _Find_last_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <forward_iterator _It, sentinel_for<_It> _Se, class _Ty, class _Pj = identity>
+            requires indirect_binary_predicate<ranges::equal_to, projected<_It, _Pj>, const _Ty*>
+        _NODISCARD constexpr subrange<_It> operator()(_It _First, _Se _Last, const _Ty& _Value, _Pj _Proj = {}) const {
+            _Adl_verify_range(_First, _Last);
+
+            if constexpr (bidirectional_iterator<_It>) {
+                auto _UFirst  = _Unwrap_iter<_Se>(_STD move(_First));
+                auto _ULast   = _Get_final_iterator_unwrapped<_It>(_UFirst, _STD move(_Last));
+                auto _UResult = _Find_last_unchecked(_STD move(_UFirst), _STD move(_ULast), _Value, _Pass_fn(_Proj));
+                return _Rewrap_subrange<subrange<_It>>(_First, _STD move(_UResult));
+            } else {
+                auto _UFirst  = _Unwrap_iter<_Se>(_STD move(_First));

```

Comment:
Since this line and 2811 are identical, we could push them up out of the branch structure.